pyproj/_crs.pyx

```diff
@@ -391,15 +500,56 @@ cdef class _CRS:
         """
         Returns
         -------
-        pyproj.CRS: The datum as a CRS.
+        Datum: The datum.
         """
-        cdef PJ *projobj = proj_crs_get_datum(self.projctx, self.projobj)
+        if self._datum is not None:
+            return self._datum
+        self._datum = Datum.create(self.projctx, self.projobj)
+        return self._datum
+
+    @property
+    def source_crs(self):
+        """
+        Returns
+        -------
+        CRS: The source CRS.
+        """
+        if self._source_crs is not None:
+            return self._source_crs
+        cdef PJ * projobj
+        projobj = proj_get_source_crs(self.projctx, self.projobj)
         if projobj == NULL:
             return None
         try:
-            return self.__class__(_to_wkt(self.projctx, projobj))
+            self._source_crs = self.__class__(_to_wkt(self.projctx, projobj))
```

Comment:
In general, it might be preferable to construct the sub CRS from an EPSG number, if available (and fallback on WKT) ?

(we could have a helper function for this, as this pattern happens in a few places)

Reply:
The wkt is usually as reliable as the epsg from my experience. Do you have a use case where it is not?

Reply:
Yes, that is true. Should have clarified why I mentioned it: now the string with which the object is created (`srs` attribute) is heavily used in the repr, and for that I find it nicer to have the epsg instead of the first few characters of the wkt. But that's maybe more a repr issue that can also be solved there. 
(and more in general, if there is a epsg number available, this is also shorter than the wkt)

Reply:
Okay, in that case I would prefer the WKT as it is more reliable. The `to_epsg` does a best guess and may not match the WKT perfectly (has a 70% confidence threshold). Though, I agree that having a cleaner srs in the repr would be nice.

Reply:
I guess you could call the `to_epsg(100)` for simplifying the `srs`.

Reply:
Yes, just wanted to mention the same that we could use a confidence for 100 for the concern that the epsg might not be a perfect match. If doing that, then at least for the "pefectly defined" CRSses this will work, but for not-fully specified ones (eg from older WKT or proj4 string), it makes sure it uses the correct wkt.

Reply:
I just made the change and removed the `EPSG` from the repr as it will be in the srs_repr and is not necessary anymore.

Reply:
I believe this is resolved. If not, feel free to add more comments.